Document that load_inline requires having a compiler installed

janeyx99 opened the pull request:
Prompted by this forum q: https://discuss.pytorch.org/t/are-the-requirements-for-using-torch-utils-cpp-extension-with-cuda-documented-anywhere/211222

Would be curious to know if we could get more precise.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137521

janeyx99 committed:
Document that additional toolchains need to be installed for load_inline

[ghstack-poisoned]

zou3519 commented on torch/utils/cpp_extension.py:
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
-        extension. This may use up too many resources on some systems. One
+        Since load_inline will just-in-time compile the source code, please ensure
+        that you have the right toolchains installed in the runtime. For example,
+        if you are loading C++, make sure a C++ compile is available. If you're loading
```
nit: compiler

zou3519 reviewed (approved):
Looks good to me but @malfet would know better if we can make this more precise

janeyx99 committed:
Update on "Document that additional toolchains need to be installed for load_inline"

Prompted by this forum q: https://discuss.pytorch.org/t/are-the-requirements-for-using-torch-utils-cpp-extension-with-cuda-documented-anywhere/211222

Would be curious to know if we could get more precise.




[ghstack-poisoned]

albanD commented on torch/utils/cpp_extension.py:
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
-        extension. This may use up too many resources on some systems. One
+        Since load_inline will just-in-time compile the source code, please ensure
+        that you have the right toolchains installed in the runtime. For example,
+        if you are loading C++, make sure a C++ compiler is available. If you're loading
```
> if you are loading C++

Why the "if", isn't this always loading c++ code?

janeyx99 replied:
okay yea

albanD commented on torch/utils/cpp_extension.py:
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
+        Since load_inline will just-in-time compile the source code, please ensure
+        that you have the right toolchains installed in the runtime. For example,
+        if you are loading C++, make sure a C++ compiler is available. If you're loading
+        a CUDA extension, you will need to additionally install the corresponding CUDA nvcc
+        and any other dependencies your code has. Compiling toolchains are not included
```
nit: I would name this "cuda toolkit (that contains nvcc and necessary headers in particular)

janeyx99 replied:
Is this cuda toolkit the same one available on conda? Cuz the forum q poster mentioned installing this and having it not be enough

janeyx99 committed:
Update on "Document that load_inline requires having a compiler installed"

Prompted by this forum q: https://discuss.pytorch.org/t/are-the-requirements-for-using-torch-utils-cpp-extension-with-cuda-documented-anywhere/211222

Would be curious to know if we could get more precise.




[ghstack-poisoned]

janeyx99 committed:
Update on "Document that load_inline requires having a compiler installed"

Prompted by this forum q: https://discuss.pytorch.org/t/are-the-requirements-for-using-torch-utils-cpp-extension-with-cuda-documented-anywhere/211222

Would be curious to know if we could get more precise.




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge
